### PR TITLE
[DATA-114 + DATA-631] Adjust test for updating & localization mode

### PR DESCRIPTION
**Pull request by kkufieta**
Adjusts tests to work with added updating and localization modes:
* https://viam.atlassian.net/browse/DATA-631
* https://viam.atlassian.net/browse/DATA-114

### CODE REVIEW

File: services/slam/builtin/cartographer_int_test.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func testCartographerPositionAndMap(t *testing.T, svc slam.Service) {
 	actualMIME, _, pointcloud, err := svc.GetMap(context.Background(), "test", "pointcloud/pcd", nil, false)
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, actualMIME, test.ShouldResemble, "pointcloud/pcd")
-	test.That(t, pointcloud.Size(), test.ShouldEqual, 200)
+	test.That(t, pointcloud.Size(), test.ShouldBeGreaterThanOrEqualTo, 100)
```
Comment (kkufieta) [inline services/slam/builtin/cartographer_int_test.go:52]: We can't assume that a point cloud will have a certain number of points. Instead, we estimate that there is a minimum amount of points added.

File: services/slam/builtin/cartographer_int_test.go [hunk 2]
```diff
@@ -115,6 +115,14 @@ func TestCartographerIntegration(t *testing.T) {
 	lastFileName := files[len(files)-1].Name()
 	test.That(t, os.Remove(name+"/data/"+lastFileName), test.ShouldBeNil)
 
+	// Remove maps so that testing in offline mode will run in mapping mode,
+	// as opposed to updating mode.
+	files, err = ioutil.ReadDir(name + "/map/")
+	test.That(t, err, test.ShouldBeNil)
+	for _, file := range files {
+		test.That(t, os.Remove(name+"/map/"+file.Name()), test.ShouldBeNil)
```
Comment (jeremyhyde-viam) [inline services/slam/builtin/cartographer_int_test.go:123]: [minor] Switch removal to os.RemoveAll(name + "/map/*")

File: services/slam/builtin/cartographer_int_test.go (hunk 1 above)
Comment (jeremyhyde-viam) [reply services/slam/builtin/cartographer_int_test.go:52]: [q] Is it not deterministic, given the same config para and dataset such that we should be able to estimate the exact number of points?

**Review by jeremyhyde-viam — COMMENTED** [review]
Simply file removal and question regarding how deterministic cartographer should be in terms of pointcloud number

**Review by jeremyhyde-viam — APPROVED** [review]
LGTM, consider adding a comment with the expected number so we can monitor and larger changes in it

**Pull request merged**